tests/syntax-tests/create_highlighted_versions.py

```diff
@@ -31,16 +43,24 @@ def create_highlighted_versions(output_basepath):
             env.pop("BAT_TABS", None)
             env["COLORTERM"] = "truecolor"  # make sure to output 24bit colors
 
-            bat_output = subprocess.check_output(
-                ["bat"] + BAT_OPTIONS + [source], stderr=subprocess.PIPE, env=env,
-            )
-
-            source_dirname = path.basename(path.dirname(source))
+            source_dirpath = path.dirname(source)
+            source_dirname = path.basename(source_dirpath)
             source_filename = path.basename(source)
 
-            if source_filename == "LICENSE.md":
+            if source_filename in SKIP_FILENAMES:
                 continue
 
+            options = copy.deepcopy(BAT_OPTIONS)
+            files = [f for (_, _, f) in os.walk(source_dirpath)]
+            # If a directory is empty, `files` could possibly be 0-length
+            if len(files) > 0 and "bat_options" in files[0]:
```

Comment:
That seems wrong. Why would `bat_options` be the first file to be listed?

Reply:
> I think we can directly check if that file exists. No need to os.walk, right?

Looks like yes. Sorry, python isn't my main language.